### PR TITLE
Upgrade invitations service (Pydantic v2)

### PR DESCRIPTION
**Pull request by giancarloromeo**
## What do these changes do?
Upgrade the invitation service (Pydantic v2).

## Related issue/s

- resolves #4481 

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
:tada:

File: services/invitations/src/simcore_service_invitations/services/invitations.py [hunk 1]
```diff
@@ -35,36 +39,18 @@ def serialize(cls, model_obj: InvitationContent) -> str:
...
-                "alias": "e",
-            },
-            "product": {
-                "alias": "p",
-            },
-            "created": {
-                "alias": "c",
-            },
-        }
+        alias_generator=_to_initial,
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/invitations/src/simcore_service_invitations/services/invitations.py:49]: The iplemenation you did of `_to_initial` has a very high risk of returning the same alias for two different (future) field names. Even if it is less efficient, let's rewrite it in a more explicit why using some sort of dict

```python
aliases_map = {
            "issuer": {
                "alias": "i",
            },
            "guest": {
                "alias": "g",
            },
            "trial_account_days": {
                "alias": "t",
            },
            "extra_credits_in_usd": {
                "alias": "e",
            },
            "product": {
                "alias": "p",
            },
            "created": {
                "alias": "c",
            },
        }

_create_initial_alias = lambda field_name: aliases_map[field_name]["alias"]
```

Even `"alias" is not needed

**Review by pcrespov — CHANGES_REQUESTED** [review]
thx. Please check the alias generator

**Review by pcrespov — APPROVED** [review]
thx

File: services/invitations/src/simcore_service_invitations/services/invitations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (giancarloromeo) [reply services/invitations/src/simcore_service_invitations/services/invitations.py:49]: I added the `assert` statement as test to catch alias repetition early and have a clean separation between validation and model logic.

**Pull request merged**